be/src/exec/pipeline/crossjoin/cross_join_context.cpp

```diff
@@ -0,0 +1,41 @@
+#include "exec/pipeline/crossjoin/cross_join_context.h"
+
+#include <algorithm>
+#include <numeric>
+
+#include "common/global_types.h"
+#include "exec/pipeline/runtime_filter_types.h"
+#include "exec/vectorized/cross_join_node.h"
+#include "exprs/expr.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks::pipeline {
+Status CrossJoinContext::_init_runtime_filter(RuntimeState* state) {
+    vectorized::ChunkPtr one_row_chunk = nullptr;
+    size_t num_rows = 0;
+    for (auto& chunk_ptr : _build_chunks) {
+        if (chunk_ptr != nullptr) {
+            if (chunk_ptr->num_rows() == 1) {
+                one_row_chunk = chunk_ptr;
+            }
+            num_rows += chunk_ptr->num_rows();
+        }
+    }
+    // build runtime filter for cross join
+    if (num_rows == 1) {
```

Comment:
if  num_rows >1, then RuntimeFilterHub::set_collector is not invoked. in such a case, the left side pipeline driver would in PRE_CONDITION_BLOCK state forever, so here should also invoke set_collector with a trivial rf to unblock left side PipelineDriver to run, tleft side PipelineDriver check the trivial rf and ignore it.

Reply:
you are right it blocked. 
eg:
```
 select count(*) from lineorder join dates where lo_orderdate >= d_datekey and( d_datekey = 19980802 or d_datekey=19970802);
```